sdk/src/asset_handlers/png_io.rs

```diff
@@ -384,6 +384,44 @@ impl AssetIO for PngIO {
 
         Ok(positions)
     }
+
+    fn remove_cai_store(&self, asset_path: &Path) -> Result<()> {
+        // get png byte
+        let mut png_buf = std::fs::read(asset_path).map_err(|_err| Error::EmbeddingError)?;
+
+        let mut cursor = Cursor::new(png_buf);
+        let ps = get_png_chunk_positions(&mut cursor)?;
+
+        // get back buffer
+        png_buf = cursor.into_inner();
+
+        /*  splice in new chunk.  Each PNG chunk has the following format:
+                chunk data length (4 bytes big endian)
+                chunk identifier (4 byte character sequence)
+                chunk data (0 - n bytes of chunk data)
+                chunk crc (4 bytes in crc in format defined in PNG spec)
+        */
+
+        // erase existing
+        let empty_buf = Vec::new();
+        let mut iter = ps.into_iter();
+        if let Some(existing_cai) = iter.find(|pcp| pcp.name == CAI_CHUNK) {
+            // replace existing CAI
+            let start = usize::value_from(existing_cai.start)
+                .map_err(|_err| Error::BadParam("value out of range".to_string()))?; // get beginning of chunk which starts 4 bytes before label
```

Comment:
I'm not sure `BadParam` (typically error on the part of the caller) makes sense here. This is really saying the existing file is has faulty data in it, right?

Reply:
Or the start cannot fit into a usize.

Reply:
Hey, same comment I made above. I think we should identify these as invalid bmff content.